File: implementations/js/runner.js
var ndarray = require('ndarray');
var show = require('ndarray-show');

function runner(scale) {
    if (typeof performance === "undefined") {
       performance = Date;
    }

     var Lx=.05, Ly=.04, Lz=.03; // Cavity dimensions in meters.
     var Nx=25, Ny=20, Nz=15;    // Number of cells in each direction.

     // Because norm isn't currently supported,
     // nrm=norm([Nx/Lx Ny/Ly Nz/Lz]) is plugged in.
     var nrm=866.0254;

     var Nt=scale*200; // Number of time steps.

     // Run kernel and measure time for core computation
     var startTime = performance.now();
     for (var i = 0; i < scale; ++i) {
         var output = fdtd(Lx,Ly,Lz,Nx,Ny,Nz,nrm,Nt);
     }
     var elapsedTime = (performance.now() - startTime) / 1000;

     var ADJUST = 1000;
     var checksum = fletcher_sum_ndarray(ndarray([
         output.Ex,
         output.Ey,
         output.Ez,
         output.Hx,
         output.Hy,
         output.Hz
     ].map(function (x) {
         ops.mulseq(x, ADJUST);
         ops.flooreq(x);
         return fletcher_sum_ndarray(x);
     })));

     console.log('{' +
     '    "time": ' + elapsedTime +
     ',   "output": ' + checksum +
     '}');
}

File: implementations/matlab/runner.m
function runner(scale)
%%
%% Driver for 3D FDTD of a hexahedral cavity with conducting walls.
%%

% Parameter initialization.
Lx=.05; Ly=.04; Lz=.03; % Cavity dimensions in meters.
Nx=25; Ny=20; Nz=15; % Number of cells in each direction.

% Because norm isn't currently supported,
% nrm=norm([Nx/Lx Ny/Ly Nz/Lz]) is plugged in.
nrm=866.0254;

Nt=scale*200; % Number of time steps.

tic();
[Ex, Ey, Ez, Hx, Hy, Hz, Ets]=fdtd(Lx, Ly, Lz, Nx, Ny, Nz, nrm, Nt);
elapsed = toc();

ADJUST = 1000;
checksum = fletcherSum([fletcherSum(floor(ADJUST*Ex)),
    fletcherSum(floor(ADJUST*Ey)),
    fletcherSum(floor(ADJUST*Ez)),
    fletcherSum(floor(ADJUST*Hx)),
    fletcherSum(floor(ADJUST*Hy)),
    fletcherSum(floor(ADJUST*Hz))]);

disp('{');
disp('"time":');
disp(elapsed);
disp(', "output":');
disp(checksum);
disp('}');

end
